### PR TITLE
Log clicks on Announcements on script and course overview pages

### PR DESCRIPTION
**Pull request by bencodeorg**
Logs to Firehose any time a user clicks on a button on an announcement on script and course overview pages.

A couple notes:
- ~~"injects" analytics data to be logged to Firehose into announcements data in the script and course overview components. Alternative was to use a version of the Notification component that is connected to the redux store, but I thought it was cleaner this way.~~
- analytics data from the course and script overview pages are currently the same for each Notification, so we use a single prop when creating the `Announcements` component instead of injecting the analytics data into each Announcement.
- A nice to have for this item was to add GA logging as well, which I did not do. We have some [existing functionality](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/Notification.jsx#L188) on `Notification` components to allow logging to GA, but it involves providing an analyticId, which I didn't see a clean way providing an ID for each announcement.

Note for anyone reading that I didn't realize for a while while working on this: logging of script ID and user ID are [actually built into our Firehose client](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/lib/util/firehose.js#L171), but only on pages where we have access to `pageConstants`, which is not the case on these pages (I think `pageConstants` is available when you're working on a level? Not 100% sure).

## Links

- jira ticket: [LP-1576](https://codedotorg.atlassian.net/browse/LP-1576)

## Testing story

I tested this manually on /s/csp1-2020, which has an announcement. I also added/modified tests to the Notification and Announcement test files to confirm at least that components that have these new props are able to render.

## PR Checklist:

- [x] Tests provide adequate coverage
- [x] Privacy and Security impacts have been assessed
- [x] Code is well-commented
- [x] New features are translatable or updates will not break translations
- [x] Relevant documentation has been added or updated
- [x] User impact is well-understood and desirable
- [x] Pull Request is labeled appropriately
- [x] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/test/unit/templates/NotificationTest.js [hunk 1]
```diff
@@ -77,7 +81,7 @@ describe('Notification', () => {
         buttonText={announcement.buttonText}
         buttonLink={announcement.link}
         newWindow={true}
-        analyticId={announcement.id}
+        analyticsData={announcement.analyticsData}
```
Comment (bencodeorg) [inline apps/test/unit/templates/NotificationTest.js]: analyticId was undefined, so I just removed it.

File: apps/test/unit/code-studio/components/progress/AnnouncementsTest.jsx [hunk 2]
```diff
@@ -19,7 +21,7 @@ const defaultProps = {
 describe('Announcements', () => {
   it('does not show Notifications if no announcements', () => {
     const wrapper = shallow(<Announcements {...defaultProps} />);
-    assert.equal(wrapper.find('Connect(Notification)').length, 0);
+    assert.equal(wrapper.find(Notification).length, 0);
```
Comment (bencodeorg) [inline apps/test/unit/code-studio/components/progress/AnnouncementsTest.jsx:28]: These changes are unrelated cleanup.

File: apps/src/templates/Notification.jsx [hunk 3]
```diff
@@ -145,9 +146,10 @@ class Notification extends Component {
     dismissible: PropTypes.bool.isRequired,
     onDismiss: PropTypes.func,
     newWindow: PropTypes.bool,
-    // analyticId is only posted when a primary button is provided.
+    // analyticId and analyticsData are only used when a primary button is provided.
     // It's not used by the array of buttons.
     analyticId: PropTypes.string,
+    analyticsData: PropTypes.object,
```
Comment (bencodeorg) [inline apps/src/templates/Notification.jsx]: Maybe I should make this something like `firehoseData`, and change the existing `analyticId` to `googleAnalyticsId`?

File: apps/src/templates/courseOverview/CourseOverview.js [hunk 4]
```diff
@@ -212,7 +224,7 @@ class CourseOverview extends Component {
         )}
         {isSignedIn && (
           <Announcements
-            announcements={this.props.announcements}
+            announcements={this.getAnnouncementsWithAnalyticsData()}
```
Comment (jamescodeorg) [inline apps/src/templates/courseOverview/CourseOverview.js]: Is the analytics data specific to each announcement or the same for all announcements?  (Wondering why we're splicing it into each announcement vs. passing a separate prop to Announcements.)

File: apps/src/templates/Notification.jsx (hunk 3 above)
Comment (jamescodeorg) [reply apps/src/templates/Notification.jsx]: If it's not too much trouble, it'd probably be helpful to rename.

File: apps/src/templates/courseOverview/CourseOverview.js (hunk 4 above)
Comment (bencodeorg) [reply apps/src/templates/courseOverview/CourseOverview.js]: mmm good point, not in this case. I guess it would make it a bit more flexible to allow for information specific to each announcement to be included, but if we want to keep it simple for the case we're using it for now I can make it its own prop.

File: apps/src/templates/Notification.jsx (hunk 3 above)
Comment (bencodeorg) [reply apps/src/templates/Notification.jsx]: It looks like `analyticId` is only used a couple of places, so should be pretty easy.

File: apps/src/templates/courseOverview/CourseOverview.js (hunk 4 above)
Comment (bencodeorg) [reply apps/src/templates/courseOverview/CourseOverview.js]: @jamescodeorg prefer that I don't inject the analytics data into the announcement?

Comment (jamescodeorg) [reply apps/src/templates/courseOverview/CourseOverview.js]: If we had announcement-specific data that we wanted to put in the analytics, where would it come from?  Would we just put it on announcement shape?

I guess this is the thing that feels weird about this -- if the analytics data is per-announcement, shouldn't we just add it to the announcement shape (instead of injecting it in ScriptOverviewHeader)?

If we don't think that we'll have per-announcement analytics data, then I think passing it as a single prop to Announcements would be easier?

Comment (bencodeorg) [reply apps/src/templates/courseOverview/CourseOverview.js]: > If we had announcement-specific data that we wanted to put in the analytics, where would it come from? Would we just put it on announcement shape?
> 
> I guess this is the thing that feels weird about this -- if the analytics data is per-announcement, shouldn't we just add it to the announcement shape (instead of injecting it in ScriptOverviewHeader)?
> 
> If we don't think that we'll have per-announcement analytics data, then I think passing it as a single prop to Announcements would be easier?

I don't have a strong preference -- I'll make analyticsData it's own prop :).

**Review by jamescodeorg — APPROVED** [review]
